ext/web/lib.rs

```diff
@@ -314,46 +316,32 @@ impl Resource for TextDecoderResource {
   }
 }
 
-#[derive(Serialize)]
-#[serde(rename_all = "camelCase")]
-struct EncodeIntoResult {
-  read: usize,
-  written: usize,
-}
-
-#[op]
+#[op(v8)]
 fn op_encoding_encode_into(
-  input: String,
+  scope: &mut v8::HandleScope,
+  input: serde_v8::Value,
   buffer: &mut [u8],
-) -> EncodeIntoResult {
-  // Since `input` is already UTF-8, we can simply find the last UTF-8 code
-  // point boundary from input that fits in `buffer`, and copy the bytes up to
-  // that point.
-  let boundary = if buffer.len() >= input.len() {
-    input.len()
-  } else {
-    let mut boundary = buffer.len();
-
-    // The maximum length of a UTF-8 code point is 4 bytes.
-    for _ in 0..4 {
-      if input.is_char_boundary(boundary) {
-        break;
-      }
-      debug_assert!(boundary > 0);
-      boundary -= 1;
-    }
-
-    debug_assert!(input.is_char_boundary(boundary));
-    boundary
+  out_buf: &mut [u8],
+) {
+  let s = v8::Local::<v8::String>::try_from(input.v8_value).unwrap();
+  assert!(out_buf.len() % 4 == 0);
+  // SAFETY: `out_buf` is guaranteed to be aligned to 4 bytes.
+  let out_buf: &mut [u32] = unsafe {
+    std::slice::from_raw_parts_mut(
+      out_buf.as_mut_ptr() as *mut u32,
+      out_buf.len() / std::mem::size_of::<u32>(),
+    )
```

Comment:
Alternatively, you could use `[T]::align_to_mut` (optionally then assert that the ends are empty)?
- <https://doc.rust-lang.org/std/primitive.slice.html#method.align_to_mut>
Then .try_into `&mut [u32]` -> `&mut [u32; 2]`?

Reply:
umm tbh the code is much clear rn... `align_to_mut` is a weird name and is doing a bunch of weird stuff https://doc.rust-lang.org/src/core/slice/mod.rs.html#3506 

plus friendly-looking abstractions are the main reason why encodeInto was slow in the first place 😉

Reply:
Fair enough!
Although, if that level of optimisation matters, I'd definitely still recommend the up-front `out_buf.try_from()?`, as it avoids the two panic branches, one for each index accessed.